src/mds/CDir.h

```diff
@@ -574,11 +576,22 @@ class CDir : public MDSCacheObject, public Counter<CDir> {
     if (is_auth()) 
       get_dist_spec(dist, whoami);
 
-    encode(frag, bl);
```

Comment:
Sorry, one last change. Let's make:

`static void encode_empty_dirstat(...)`

into

`static void encode_dirstat(...)`

(So, there are two methods, one static and the other not.) Call the static method from non-static one using a `DirStat` derived from the object.. The code in the Server should look similar to what you did for the null `LeaseStat`.

Reply:
The point of me asking this is that I don't want duplicate code doing the encoding.

Reply:
@batrick Yes, I'm working on this change with DirStat now. Thanks.